pkg/plugin/plugin.go

```diff
@@ -296,10 +392,78 @@ func (g *GlooMatchedRouteTable) matchRoutes(logCtx *logrus.Entry, rollout *v1alp
 					CanaryOrPreviewDestination: canary,
 				},
 			}
-			logCtx.Debugf("adding destination %+v", dest)
 			g.HttpRoutes = append(g.HttpRoutes, dest)
 		}
 	} // end range httpRoutes
 
 	return nil
 }
+
+func buildGlooHTTPRoute(name string, matcher *solov2.HTTPRequestMatcher, canaryDestination *SimpleDestinationReference) *GlooMatchedHttpRoutes {
+	return &GlooMatchedHttpRoutes{
+		HttpRoute: &networkv2.HTTPRoute{
+			Name: name,
+			Matchers: []*solov2.HTTPRequestMatcher{
+				matcher,
+			},
+			ActionType: &networkv2.HTTPRoute_ForwardTo{
+				ForwardTo: &networkv2.ForwardToAction{
+					Destinations: []*solov2.DestinationReference{
+						{
+							RefKind: &solov2.DestinationReference_Ref{
+								Ref: &solov2.ObjectReference{
+									Name:      canaryDestination.Reference.Name,
+									Namespace: canaryDestination.Reference.Namespace,
+								},
+							},
+							Port: &solov2.PortSelector{
+								Specifier: &solov2.PortSelector_Number{
+									Number: canaryDestination.Port.Number,
+								},
+							},
```

Comment:
Do we need this port logic to be conditional based on port being a number or a name? If we're not supporting Name in this first cut, we might need to call that out in the readme explicitly as a known limitation.

Reply:
GoodPoint. I don't think we have a validation option here at submit so we can have 1 supersede the other or just fully error out if both are provided. Not sure which should supersede. Either way we go this caveat should be documented I guess.

Reply:
If we can't prevent the user from attempting to submit both with something like a `oneof` in the schema, then I think we should just error out during that validation step if they do submit both. Quickest feedback loop and easiest to understand as a user IMHO, rather than trying to reason about a priority hierarchy, which we'd also have to document with examples etc.

Reply:
Fixed this a different way.